### DISCUSSION TITLE
Extracting an Ordinal's artist/creator from on-chain data

### DISCUSSION THREAD

**Discussion by nosuic** [ordinals/ord#3663] [Q&A] [answered]
Hi all, 

I am doing a research project on ordinals (I have [published research on NFTs previously](https://www.mdpi.com/2076-3417/11/21/9931)) and I am trying to figure out the information schema behind ordinals and how this could be extracted from on-chain data. Could someone please explain:

1. Who is considered the creator of an ordinal on-chain? Is it the wallet which initiated the reveal transaction for example? 
2. (Similarly) How could I extract the creator's wallet from on-chain data for a given transaction concerning an ordinal (e.g. [95402bb99ee2c6b955363491a74f3989c17ae05a2bde56810e8077ce9ccecfc6](https://www.blockchain.com/explorer/transactions/btc/95402bb99ee2c6b955363491a74f3989c17ae05a2bde56810e8077ce9ccecfc6))

If you could please shed any light on that, I would greatly appreciate it. Thanks!

**Comment by cryptoni9n** [comment]
Hi nosuic,

1. the creator of the ordinal inscription is the inscriber.  they are the input address in the genesis transaction of the ordinal.  [This ordinal,](http://ordinals.com/inscription/a356f858f97437f7752c78c6638f076ba00f6a462d246bb5115be8b6371e2fdbi0) for example, was inscribed by address bc1pnwcs5t0aga7uvhf29659xlfhx870k0vw69nsxcwvwtl38qjmn8jq3agqay and they would be considered the creator.

2. for this example, you would get the [reveal transaction](http://ordinals.com/tx/a356f858f97437f7752c78c6638f076ba00f6a462d246bb5115be8b6371e2fdb) for the inscription genesis, the go to [the input](http://ordinals.com/output/700a095e0f1f29a1fc80f2326e2e6242bfbae6f81d7f09ba913e2a006a42061a:2) and grab the address that is displayed there.

I hope this helps!

**Reply by nosuic** [reply]
Hi @cryptoni9n,

Thanks so much for taking the time to reply, that's very helpful. 

I have been looking to detect the artist rather than the minter (which I realise was not clarified in my question) and I discovered that provenance can be traced sometimes through parent inscriptions and other times through referenced recursive submodules. And I have been wondering:

1. How could I use ord (i.e. latest ord release on Bitcoin Core) to extract the parent of an inscription?
2. Similarly how could I extract the recursive submodule(s) of an inscription?

Thanks again!

**Reply by cryptoni9n** [reply]
>     1. How could I use ord (i.e. latest ord release on Bitcoin Core) to extract the parent of an inscription?
> 
>     2. Similarly how could I extract the recursive submodule(s) of an inscription?
>
You could use the [inscription endpoint](https://docs.ordinals.com/inscriptions/recursion.html) for both cases, like this: 
1) `curl.exe -s -H "Accept: application/json" "http://localhost/inscription/681b5373c03e3f819231afd9227f54101395299c9e58356bda278e2f32bef2cdi0" | jq '.parents'"`
and 
2) `curl.exe -s -H "Accept: application/json" "http://localhost/inscription/681b5373c03e3f819231afd9227f54101395299c9e58356
bda278e2f32bef2cdi0" | jq`

**Reply by nosuic** [reply]
Thank you, that's great. I will try these out and let you know if I have any follow up questions. If I get it to work, I will mark this answer and close the question. Thanks again!

**Reply by nosuic** [reply]
Hi @cryptoni9n! Following your advice I was able to successfully extract data using `ord` on Bitcoin node. The command for extracting parent-child relationships worked great! 

However I have been unable to extract information on recursive modules. Running: `curl -s -H "Accept: application/json" "http://localhost/inscription/71057820" | jq >output3.json` as suggested  (#2 above), returns the following which does not include any indication for the inscription's recursive modules (you can see those here: https://ordiscan.com/inscription/71057820).

```
{
  "address": "bc1payw0jhpxzrha0x4xmc5lfp2elmekg7xg2yrklvvd7m8dtr7xjmmsvjgv0c",
  "charms": ["vindicated"],
  "children": [],
  "content_length": 186,
  "content_type": "text/html;charset=utf-8",
  "effective_content_type": "text/html;charset=utf-8",
  "fee": 2035,
  "height": 844868,
  "id": "3fd1555a7d2debd7d578ed3b6589ba184629257c4580fe56c3863147837fc7f6i0",
  "next": "3f76fc72e1eb72adf7cf6de391d000d1d8d8724a8f99a6b102130e9ee845cf13i0",
  "number": 71057820,
  "parents": [],
  "previous": "598f976adc5b371ba91854178f28e297d15394eaded3ed9b8a22272aa01c18f2i0",
  "rune": null,
  "sat": null,
  "satpoint": "3fd1555a7d2debd7d578ed3b6589ba184629257c4580fe56c3863147837fc7f6:0:0",
  "timestamp": 1716524914,
  "value": 546
}
```

Should I use some other endpoint to extract the recursive modules of a given inscription?

Thanks again for all the help!

**Reply by cryptoni9n** [reply, marked as answer]
Hi @nosuic - yeah, that's a bit more complicated.

You can hit the content endpoint to read the script of an inscription:

`curl.exe -s -H "Accept: application/json" "http://localhost/content/3fd1555a7d2debd7d578ed3b6589ba184629257c4580fe56c3863147837fc7f6i0"`

Then, you'll have to have a script that can parse the output for other inscription ID's.  So for example, the command above returns the script for the example:

`<script src="/content/d91650a911e27c5402ffd45893b17023e0ede8cb6912aa11fee0bbb264de2bfdi0" data-p="991e410c58fcdc900553513743cdca3086049b7c5a03bce7f165873d0e3891f9i0,s,3,64,1,l">`

You'd need to then parse that data to pull out the two inscription ID's - they are the recursive modules.
